starlingcaptureapi/file_util.py

```diff
@@ -56,3 +58,37 @@ def digest_sha256(self, file_path):
                 hasher.update(byte_block)
             return hasher.hexdigest()
         # TODO: handle error (image not found, etc.)
+
+    def timestamp(self, file_path, ts_file_path, timeout=5, min_cals=2):
+        """Creates a opentimestamps file for the given file.
+
+        Args:
+            file_path: path to file
+            ts_file_path: output path for opentimestamps file (.ots)
+            timeout: Timeout before giving up on a calendar
+            min_cals: timestamp is considered done if at least this many calendars replied
+
+        Raises:
+            any file I/O errors
+            Exception if errors are encountered during processing
+        """
+
+        with open(file_path, "rb") as inp, open(ts_file_path, "wb") as out:
+            proc = subprocess.run(
```

Comment:
Same comment as in the IPFS CID PR (https://github.com/starlinglab/starling-integrity-api/pull/64/files#r824242498) about maybe extracting this into a separate lib for running various CLIs and raising their errors.

Reply:
Agree to move the implementation-specific function away from `FileUtil`, then call from `FileUtil`. Otherwise `FileUtil` gets really hard to read.

Shall we have an:
- opentimestamps.py
- ipfs.py

Or perhaps if we have a `cli.py` that contains `OpenTimestamps` and `Ipfs` classes, that makes it more likely we will have consistent exception raising? 🤔